src/config.rs

```diff
@@ -546,6 +546,19 @@ fn config_from_env() -> EnvConfig {
     EnvConfig { cache }
 }
 
+fn config_file(env_var: &str, leaf: &str) -> PathBuf {
+    if let Some(env_value) = env::var_os(env_var) {
+        return env_value.into();
+    }
+    let dirs =
+        ProjectDirs::from("", ORGANIZATION, APP_NAME).expect("Unable to get config directory");
+    let path = dirs.preference_dir().join(leaf);
+    if path.exists() {
+        return path;
+    }
+    dirs.config_dir().join(leaf)
+}
```

Comment:
Maybe we fast-path the new path if it already exists, if not we check the old path and return that if it exists, and return the new path if neither exist yet.

Another option is to migrate the config file and/or log a warning somewhere saying we're moving the config file and we're deprecating use of the old one.